### DISCUSSION TITLE
Is there any whitepaper / development wiki for contributors?

### DISCUSSION THREAD

**Discussion by dmknght** [evilsocket/opensnitch#423] [Development]
I think it is really helpful to have something like development documentation so any user can read and understand about code structure -> audit the source code and contribute more.
For me i would like to understand more about how python gui is connecting with daemon so i can try improve GUI or create alternative GUI :D

**Comment by gustavo-iniguez-goya** [comment]
There's no such thing I'm afraid.

I've documented many functions, maybe you can get an idea by reading the code.

On the other hand, the cli tool also acts as a server, maybe by reading it you can get some ideas:
https://github.com/gustavo-iniguez-goya/opensnitch/blob/f1d406f19b75248702b1e8b5480c9b227c3d4de5/server/api/api.go

I think these topics are better discussed on the Discussion forums, feel free to close this issue and open a new thread.

**Comment by dmknght** [comment]
> There's no such thing I'm afraid.
> 
> I've documented many functions, maybe you can get an idea by reading the code.
> 
> On the other hand, the cli tool also acts as a server, maybe by reading it you can get some ideas:
> https://github.com/gustavo-iniguez-goya/opensnitch/blob/f1d406f19b75248702b1e8b5480c9b227c3d4de5/server/api/api.go
> 
> I think these topics are better discussed on the Discussion forums, feel free to close this issue and open a new thread.

Thank you! That is a `--` for me because i really want to know how OpenSnitch works :( anyway i'm having some ideas for the GUI and i think modify the current code is best way :D
